app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php

```diff
@@ -878,6 +915,23 @@ protected function doAddWebsiteNamesToResult()
         return $this;
     }
 
+    /**
+     * Adding information about stock item to each product.
+     *
+     * @return $this
+     * @since 101.1.0
+     */
+    protected function doAddStockDataToResult()
```

Comment:
This approach marked as deprecated
```
/**
 * @deprecated 100.2.0 Stock Item as a part of ExtensionAttributes is deprecated
 * @see StockItemInterface when you want to change the stock data
 * @see StockStatusInterface when you want to read the stock data for representation layer (storefront)
 * @see StockItemRepositoryInterface::save as extension point for customization of saving process
 */
```
https://github.com/magento/magento2/blob/4b5626478af59cc427d76024c459bd3109ec91b9/app/code/Magento/CatalogInventory/Model/Plugin/AfterProductLoad.php#L15